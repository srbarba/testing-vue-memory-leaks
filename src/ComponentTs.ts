import Vue from 'vue'
import { mapGetters } from 'vuex'

export default Vue.extend({
  computed: {
    ...mapGetters(['userName']),
  },
  render(h) {
    return h('div', `Hello, ${this.userName}`)
  },
})
