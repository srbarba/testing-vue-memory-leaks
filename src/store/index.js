export default {
  state: () => ({
    user: {
      name: undefined,
    },
  }),
  getters: {
    userName: (state) => state.user.name,
  },
  mutations: {
    setUser(state, user) {
      state.user = user
    },
  },
  actions: {},
}
