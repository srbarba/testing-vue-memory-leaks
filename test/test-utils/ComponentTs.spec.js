import Vuex from 'vuex'
import { createLocalVue, mount } from '@vue/test-utils'
import ComponentTs from '../../src/ComponentTs'
import store from '../../src/store'

describe('ComponentTs', () => {
  it('renders correctly', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    const wrapper = mount(ComponentTs, {
      localVue,
      store: new Vuex.Store(store),
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
