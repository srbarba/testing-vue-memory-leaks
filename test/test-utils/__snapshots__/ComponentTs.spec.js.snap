// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentTs renders correctly 1`] = `"<div>Hello, undefined</div>"`;
